causalml/inference/iv/drivlearner.py

```diff
@@ -0,0 +1,864 @@
+
+    A DRIV-learner estimates endogenous treatment effects for compliers with machine learning models.
+
+    Details of DR-learner are available at Kennedy (2020) (https://arxiv.org/abs/2004.14497).
+    The DR moment condition for LATE comes from Chernozhukov et al (2018) (https://academic.oup.com/ectj/article/21/1/C1/5056401).
+    """
+
+    def __init__(
+        self,
+        learner=None,
+        control_outcome_learner=None,
+        treatment_outcome_learner=None,
+        treatment_effect_learner=None,
+        ate_alpha=0.05,
+        control_name=0,
+    ):
+        """Initialize a DR-learner.
+
+        Args:
+            learner (optional): a model to estimate outcomes and treatment effects in both the control and treatment
+                groups
+            control_outcome_learner (optional): a model to estimate outcomes in the control group
+            treatment_outcome_learner (optional): a model to estimate outcomes in the treatment group
+            treatment_effect_learner (optional): a model to estimate treatment effects in the treatment group. It needs
+                to take `sample_weight` as an input argument in `fit()`.
+            ate_alpha (float, optional): the confidence level alpha of the ATE estimate
+            control_name (str or int, optional): name of control group
+        """
+        assert (learner is not None) or (
+            (control_outcome_learner is not None)
+            and (treatment_outcome_learner is not None)
+            and (treatment_effect_learner is not None)
+        )
+
+        if control_outcome_learner is None:
+            self.model_mu_c = deepcopy(learner)
+        else:
+            self.model_mu_c = control_outcome_learner
+
+        if treatment_outcome_learner is None:
+            self.model_mu_t = deepcopy(learner)
+        else:
+            self.model_mu_t = treatment_outcome_learner
+
+        if treatment_effect_learner is None:
+            self.model_tau = deepcopy(learner)
+        else:
+            self.model_tau = treatment_effect_learner
+
+        self.ate_alpha = ate_alpha
+        self.control_name = control_name
+
+        self.propensity_1 = None
+        self.propensity_0 = None
+        self.propensity_assign = None
+
+    def __repr__(self):
+        return (
+            "{}(control_outcome_learner={},\n"
+            "\ttreatment_outcome_learner={},\n"
+            "\ttreatment_effect_learner={})".format(
+                self.__class__.__name__,
+                self.model_mu_c.__repr__(),
+                self.model_mu_t.__repr__(),
+                self.model_tau.__repr__(),
+            )
+        )
+
+    def fit(
+        self, X, assignment, treatment, y, p=None, pZ=None, seed=None, calibrate=True
+    ):
+        """Fit the inference model.
+
+        Args:
+            X (np.matrix or np.array or pd.Dataframe): a feature matrix
+            assignment (np.array or pd.Series): a (0,1)-valued assignment vector
```

Comment:
What's `assignment`? Could you explain little more?

Reply:
I added explanations in the docstring.